lms/djangoapps/certificates/tests/test_queue.py

```diff
@@ -73,6 +76,51 @@ def test_no_create_action_in_queue_for_html_view_certs(self):
         self.assertEqual(certificate.status, CertificateStatuses.downloadable)
         self.assertIsNotNone(certificate.verify_uuid)
 
+    @ddt.data('credit', 'verified', 'honor', 'audit')
+    def test_add_cert_with_different_modes(self, mode):
+        """Test if enrollment mode is verified or credit along with valid
+        software-secure verification than verified certificate should be generated.
+        """
+        user_2 = UserFactory.create()
+        CourseEnrollmentFactory(
+            user=user_2,
+            course_id=self.course.id,
+            is_active=True,
+            mode=mode,
+        )
+        SoftwareSecurePhotoVerificationFactory.create(user=user_2, status='approved')
+        with patch('courseware.grades.grade', Mock(return_value={'grade': 'Pass', 'percent': 0.75})):
+            with patch.object(XQueueInterface, 'send_to_queue') as mock_send:
+                mock_send.return_value = (0, None)
+                self.xqueue.add_cert(user_2, self.course.id)
+
+        # Verify that the task was sent to the queue with the correct callback URL
+        self.assertTrue(mock_send.called)
+        __, kwargs = mock_send.call_args_list[0]
+        actual_header = json.loads(kwargs['header'])
+        self.assertIn('https://edx.org/update_certificate?key=', actual_header['lms_callback_url'])
+
+        # Verify that the task was sent to the queue with the correct template
+        actual_body = json.loads(kwargs['body'])
+
+        certificate = GeneratedCertificate.objects.get(user=user_2, course_id=self.course.id)
+        if mode in ['credit', 'verified']:
```

Comment:
Same as above. A conditional in a test usually indicates you should write a separate test case (and use shared methods to avoid duplicating code).